### DISCUSSION TITLE
Rewrites don't work as expected

### DISCUSSION THREAD

**Discussion by stormi99** [vercel/community#6302] [Help]
### Summary

Hello, I am trying to redirect "main-website.com/restaurants/[restaurantId]" to "restaurant-domain.com" I added the "restaurant-domain.com" to Domains and set my rewrites as below, when I go restaurant website ("restaurant-domain.com") it redirects to main website without changing the visible url on the browser that's what i want but it does not load any js or css. Everything returns 404 because its looking for assets in "restaurant-domain.com" but all the assets are under "main-website.com" How can I fix this problem? Thanks!
`rewrites() {
    return {
      beforeFiles: [
        {
          source: "/:path*",
          has: [
            {
              type: "host",
              value: "mama-mia-pizza.com",
            },
          ],
          destination: "/restaurants/clt4u19f3000713atj9w3njcw/:path*/",
        },
      ],
    };
  },`
![image](https://github.com/vercel/community/assets/132082967/a870bcf6-e8b9-45b4-ab62-8adef445fd91)
![image](https://github.com/vercel/community/assets/132082967/48384a85-ca7b-4234-9be6-3b1afe9bb061)

### Steps to Reproduce

You can create a test website on vercel and trying to redirect to a domain (xxx.vercel.app/test-site) to (yourdomain.com)

**Comment by TooTallNate** [comment]
Sounds like you want to use [`redirects` config](https://vercel.com/docs/edge-network/redirects) instead of `rewrites`.

**Comment by stormi99** [comment]
Would redirects not change to url? It is so important to keep restaurant
domain while hosting as main website.

**Reply by TooTallNate** [reply]
Redirects would indeed change the URL.

Perhaps you can try to describe at a more high level what you are trying to achieve. I seems like there's possibly an [X Y problem](https://en.wikipedia.org/wiki/XY_problem) here.

It sounds to me like you are trying to point multiple domain names to a single app (what we call "multi-tenant apps"). If that's the case, there is a [guide](https://vercel.com/guides/nextjs-multi-tenant-application) and [starter templates](https://vercel.com/templates/multi-tenant-apps) which might be useful for you.

**Reply by stormi99** [reply]
Hey man, thank you for your responses. Is it possible to point whole new domain instead of creating subdomain on that starter template? Also if you don't mind, I would like to buy a coffee for you for your help.
